src/components/MultiSelect/MultiSelect.test.tsx

```diff
@@ -0,0 +1,675 @@
+        label="MultiSelect Label"
+        variant="listbox"
+        items={items}
+        selectedItems={selectedTestItems}
+        onChange={() => null}
+        onClear={() => null}
+      />
+    );
+    expect(container.querySelector("#test")).toBeInTheDocument();
+  });
+
+  it("should initially render with a given label", async () => {
+    render(
+      <MultiSelect
+        id="test"
+        label="MultiSelect Label"
+        variant="listbox"
+        items={items}
+        selectedItems={selectedTestItems}
+        onChange={() => null}
+        onClear={() => null}
+      />
+    );
+    expect(
+      screen.getByRole("button", {
+        name: /multiselect label/i,
+      })
+    );
+  });
+
+  it("should initially render with open menu if defaultIsOpen prop is true", () => {
+    render(
+      <MultiSelect
+        id="test"
+        label="MultiSelect Label"
+        variant="listbox"
+        items={items}
+        defaultIsOpen={true}
+        selectedItems={selectedTestItems}
+        onChange={() => null}
+        onClear={() => null}
+      />
+    );
+
+    const ul = screen.getByRole("listbox");
+    const list = ul[`${Object.keys(ul)[1]}`].children;
+    expect(list).toHaveLength(6);
+  });
+
+  it("should open menu when user clicks menu button", () => {
+    render(
+      <MultiSelect
+        id="test"
+        label="MultiSelect Label"
+        variant="listbox"
+        items={items}
+        selectedItems={selectedTestItems}
+        onChange={() => null}
+        onClear={() => null}
+      />
+    );
+    userEvent.click(
+      screen.getByRole("button", {
+        name: /multiselect label/i,
+      })
+    );
+    expect(screen.getByRole("listbox")).toBeInTheDocument();
+    expect(screen.getAllByRole("checkbox")).toHaveLength(6);
+  });
+
+  it("should allow user to toggle menu by clicking menu button", () => {
+    const { container } = render(
+      <MultiSelect
+        id="test"
+        label="MultiSelect Label"
+        variant="listbox"
+        items={items}
+        selectedItems={selectedTestItems}
+        onChange={() => null}
+        onClear={() => null}
+      />
+    );
+    // When first rendered menu is not expanded
+    expect(screen.queryByRole("listbox")).toBeNull();
+
+    userEvent.click(
+      screen.getByRole("button", {
+        name: /multiselect label/i,
+      })
+    );
+
+    expect(
+      container.querySelector('[aria-expanded="true"][aria-haspopup="listbox"]')
+    ).toBeInstanceOf(HTMLElement);
+    expect(screen.getAllByRole("option")).toHaveLength(6);
+
+    userEvent.click(
+      screen.getByRole("button", {
+        name: /multiselect label/i,
+      })
+    );
+    expect(
+      container.querySelector(
+        '[aria-expanded="false"][aria-haspopup="listbox"]'
+      )
+    ).toBeInstanceOf(HTMLElement);
+    expect(screen.queryAllByRole("option")).toHaveLength(0);
+  });
+
+  it("should call onChange when an item gets selected/unselected", () => {
```

Comment:
There are no tests for the counter in the `MultiSelect` UI. Can you include the counter in this test when items are selected/unselected, or in a new test? Same for the "dialog" variant.

Reply:
There should also be a test for this counter element when it's clicked. It clears out all the selected items and closes the menu and that should be tested.